### ISSUE TITLE
[Bug] lerna exec exits early

### ISSUE THREAD

**Issue by doug-wade** [lerna/lerna#323]
I'm trying to use [lerna exec](https://github.com/lerna/lerna#exec) on beta 26 in [React Server](https://github.com/redfin/react-server) to find all of my packages that have a dependency on [superagent](https://www.npmjs.com/package/superagent) with [npm ls](https://docs.npmjs.com/cli/ls).

Here's my console output:

```
dougwade code/react-server ‹master*› » lerna exec -- npm ls superagent                                                                                                        1 ↵
Lerna v2.0.0-beta.26
react-server@0.4.6 /Users/doug.wade/code/react-server/packages/react-server
├── UNMET PEER DEPENDENCY express@>=3.x
└── superagent@1.2.0  extraneous

npm ERR! peer dep missing: express@>=3.x, required by express-state@1.2.0
npm ERR! extraneous: superagent@1.2.0 /Users/doug.wade/code/react-server/packages/react-server/node_modules/superagent
Errored while running command 'npm' with arguments 'ls superagent' in 'react-server'
Errored while running ExecCommand.execute
Command failed: npm ls superagent
dougwade code/react-server ‹master*› » babel-plugin-react-server@0.4.4 /Users/doug.wade/code/react-server/packages/babel-plugin-react-server                                  1 ↵
└── (empty)

npm ERR! code 1
babel-preset-react-server@0.4.4 /Users/doug.wade/code/react-server/packages/babel-preset-react-server
└── (empty)

npm ERR! code 1
generator-react-server@0.4.6 /Users/doug.wade/code/react-server/packages/generator-react-server
└── (empty)

npm ERR! code 1

dougwade code/react-server ‹master*› » lerna ls                                                                                                                               1 ↵
Lerna v2.0.0-beta.26
- babel-plugin-react-server
- babel-preset-react-server
- generator-react-server
- react-server
- react-server-cli
- react-server-core-middleware
- react-server-data-bundle-cache
- react-server-gulp-module-tagger
- react-server-module-tagger
```

There's a couple of things to note.  The big thing is that only four of the nine packages are visited and have output -- for instance, the "core" module, `react-server` is never visited.  

There are, however, a couple of other minor surprises: private packages aren't listed with `lerna ls`, and the command seems to exit early (or, at least, I get a command prompt halfway through execution).

**Comment by doug-wade** [comment]
For the wayfaring traveller searching these here internets for what's going on, to work around it I `for d in packages/*; do pushd $d; npm ls superagent; popd; done`

**Comment by gigabo** [comment]
Hi @doug-wade I think this is expected behavior currently.  `lerna exec` bails out when it encounters an error.  Maybe it would be better to try all packages?  Maybe an option to try all packages?

**Comment by doug-wade** [comment]
@gigabo Interesting --is the output after the first error erroneous then?  

There doesn't seem to be a way for me to suppress the error from `npm ls`, which means I don't actually have a way using lerna to discover which packages have a given dependency, which seems like something I'd want to be able to do.  Adding the ability to continue after error seems like it would work, though I could also see a lerna-specific command for reasoning about dependencies (for instance, if you were trying to [discover cycles](https://github.com/lerna/lerna/pull/302) in the dependency graph after encountering a warning).

**Comment by gigabo** [comment]
> is the output after the first error erroneous, then?

More incidental.  Scripts are run concurrently, with up to four active at once.  The output after the first error is likely from the other children that were in-flight when the error was encountered.

> a lerna-specific command for reasoning about dependencies 

Yeah, looking at dependencies across all packages in the repo is an interesting topic.

There's some more discussion in #259.

**Comment by MoOx** [comment]
Maybe related to #280.

**Comment by ryb73** [comment]
I ran into a similar use case which I think suffers from the same underlying problem. I'd like to be able to run `npm outdated` on each package to find all outdated dependencies, but npm gives a non-zero exit code if there are any outdated dependencies, so `lerna exec` stops short.

As a workaround, I've added an `outdated` script to each package which runs `"npm out | exit 0"`. The `exit 0` allows lerna to keep running the script.
